src/Native/System.Native/pal_runtimeinformation.cpp

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "pal_runtimeinformation.h"
+#include "pal_types.h"
+#include <stdio.h>
+#include <sys/utsname.h>
+
+extern "C" const char* GetUnixVersion()
+{
+    struct utsname _utsname;
+    size_t length = _UTSNAME_LENGTH * 3 + 2;
+    char * version = new char[length];
+    if (uname(&_utsname) != -1)
+    {
+        snprintf(version, length, "%s %s %s", _utsname.sysname, _utsname.release, _utsname.version);
+    }
+
+    return version;
+}
+
+/* Returns an int representing the OS Architecture:
+ 0 - x86
+ 1 - x64
+ 2 - ARM */
+extern "C" int32_t GetUnixArchitecture()
+{
+#if defined(ARM)
+    return 2;
+#elif defined(BIT64)
+    return 1;
```

Comment:
In our config files we are checking if  ```CMAKE_SYSTEM_PROCESSOR``` is ```x86_64``` string and sets the BIT64 define if true. The cmake_system_processor documentation says this is set using uname -p, which means checking for BIT64 define should mean it is actually x64. Let me know if i understood ur statement incorrectly.